src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneState = { ...state };
+  const ADD_PROPERTIES = 'addProperties';
+  const REMOVE_PROPERTIES = 'removeProperties';
+  const CLEAR = 'clear';
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === ADD_PROPERTIES) {
+      Object.assign(cloneState, action.extraData);
+      result.push({ ...cloneState });
+    } else if (action.type === REMOVE_PROPERTIES) {
+      for (const key of action.keysToRemove) {
+        delete cloneState[key];
+      }
+      result.push({ ...cloneState });
+    } else if (action.type === CLEAR) {
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions.

Reply:
Is it really necessary to change my code with a switch? 
I used only three if and I guess it's ok.